### PR TITLE
Test Ansible configuration within GitHub Actions

### PR DESCRIPTION
**Pull request by frouioui**
This pull request continues the work initiated in #20 and #16 by integrating [Molecule](https://github.com/ansible-community/molecule) to our GitHub Actions workflows.

Ansible posses a tool called Molecule easing development and testing of Ansible roles. Fortunately, our codebase already uses Ansible roles and has partially working Molecule configurations.

The new workflows shipped with this pull request are listed in the table below.

| Name        | Purpose           | Estimated time  |
| ---------- | -------------- | ---------------: |
| [Molecule Test all](https://github.com/vitessio/arewefastyet/actions/workflows/molecule-all.yml)      | Runs each role in a single instance | ~ 15 minutes |
| [Molecule Test etcd](https://github.com/vitessio/arewefastyet/actions/workflows/molecule-etcd.yml)      | Dry-run an etcd ansible role | ~ 2 minutes |
| [Molecule Test vitess_build](https://github.com/vitessio/arewefastyet/actions/workflows/molecule-vitess-build.yml) | Runs vitess_build ansible role entirely |  ~ 12 minutes |
| [Molecule Test vtctld](https://github.com/vitessio/arewefastyet/actions/workflows/molecule-vtctld.yml) | Dry-run a vtctld ansible role |    ~ 2 minutes |
| [Molecule Test vtgate](https://github.com/vitessio/arewefastyet/actions/workflows/molecule-vtgate.yml) | Dry-run a vtgate ansible role |    ~ 2 minutes |
| [Molecule Test vttablet](https://github.com/vitessio/arewefastyet/actions/workflows/molecule-vttablet.yml) | Dry-run a vttablet ansible role |    ~ 2 minutes |
| [Test all benchmarks](https://github.com/vitessio/arewefastyet/actions/workflows/test-run-all.yml) | Will allow us to run an E2E test with all the benchmarks (void for now)  |   < 1 minute |
| [Test benchmarks OLTP](https://github.com/vitessio/arewefastyet/actions/workflows/test-oltp.yml) | Will allow us to run an E2E OLTP benchmark (void for now)  |   < 1 minute |
| [Test benchmarks TPCC](https://github.com/vitessio/arewefastyet/actions/workflows/test-tpcc.yml) | Will allow us to run an E2E TPCC benchmark (void for now)  |   < 1 minute |

In addition to the GitHub Actions workflows, this pull request eases development by using Ansible's Molecule. Running the Ansible Molecule scenario located in `./ansible/roles/vitess_build/molecule/all` will spin up a ready-to-use CentOS machine with Vitess, using the same configuration as the one used when running the benchmarks in production.

Additional work such as E2E tests, adding more assertions, replacing `changed_when: false` tokens, enabling sysbench within Molecule, and so on, shall be addressed in future pull requests.

### CODE REVIEW

**Comment by frouioui** [comment]
The initial plan was to add tests using real packet devices. However, to speed up the workflow and follow ansible's best practices a bit closer, Molecule will be used to validate the different roles.

File: ansible/roles/vtgate/defaults/main.yml [hunk 1]
```diff
@@ -20,5 +20,5 @@ vtgate_grpc_port: '15306'
 
 enable_vtgate: True
 
-vtgate_query_cache_size: 0
+vtgate_query_cache_size: 1000
```
Comment (Akilan1999) [inline ansible/roles/vtgate/defaults/main.yml:23]: Interested to know why we are are changing the cache size
- Is 1000 the default for vtgate ?

Comment (frouioui) [reply ansible/roles/vtgate/defaults/main.yml:23]: The default is 5000 as shown in the code below:
https://github.com/vitessio/vitess/blob/25baa46125279255ef70bb2d78a2c3738e29f5e7/go/cache/cache.go#L76-L80

When running the Ansible roles with the inventory files we normally use (`./ansible/koz-*.yml`), we set `vttablet_query_cache_size` to a non-zero value as highlighted in this example: https://github.com/vitessio/arewefastyet/blob/4f6626497f46cf7128ec06578444caf714af7c68/ansible/koz-inventory-unsharded-test.yml#L43

However, when we run Molecule, we do not take into account these inventory files. Thus, vtgate (and vttablet) role uses their own `vttablet_query_cache_size` default value (defined in `./ansible/roles/*/defaults/*.yml).

Having a value equal to 0 causes the server to crash as defined in the below code:
https://github.com/vitessio/vitess/blob/25baa46125279255ef70bb2d78a2c3738e29f5e7/go/cache/ristretto/cache.go#L164-L165

Comment (frouioui) [reply ansible/roles/vtgate/defaults/main.yml:23]: Both vttablet and vtgate are using `vt*_query_cache_size `.

**Review by Akilan1999 — APPROVED** [review]

**Pull request merged**